synapse/cortex.py

```diff
@@ -1598,17 +1598,23 @@ async def addCoreQueue(self, qdef):
     @s_nexus.Pusher.onPush('queue:add')
     async def _addCoreQueue(self, qdef):
         iden = qdef.get('iden')
-        self.auth.reqNoAuthGate(iden)
         name = qdef.get('name')
-        if not self.multiqueue.exists(iden):
 
-            user = await self.auth.reqUser(qdef.get('creator'))
+        if self.multiqueue.exists(iden):
+            return
+
+        if self.quedefs.get(name) is not None:
+            return
```

Comment:
Can we actually make this similar to the check in _addStormMacro, where we grab the queue by name and if it exists, verify that the iden matches, to ensure this is the correct queue? Also can we have this return the qdef (either provided or existing) so the top half (addCoreQueue) can just return `_push('queue:add'` and ensure we have the right qdef?